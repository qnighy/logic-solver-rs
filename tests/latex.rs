use insta::assert_snapshot;
use std::process::Command;

#[test]
fn test_latex1() {
    assert_snapshot!(run("A"));
}

#[test]
fn test_latex2() {
    assert_snapshot!(run("A → A"));
}

#[test]
fn test_latex3() {
    assert_snapshot!(run("A → B → A"));
}

#[test]
fn test_latex4() {
    assert_snapshot!(run("(A → B → A) ∧ (A ∨ B → B ∨ A)"));
}

#[test]
fn test_latex5() {
    assert_snapshot!(run("(A → B → A) ∧ (A ∨ B → B)"));
}

#[test]
fn test_latex6() {
    assert_snapshot!(run("⇒"));
}

#[test]
fn test_latex7() {
    assert_snapshot!(run(
        "(A → A → B) → (B → B → C) → (C → C → D) → (D → D → E) → (E → E → F) → A → F"
    ));
}

fn run(s: &str) -> String {
    let output = Command::new("cargo")
        .args(&["run", "--", "-e", s, "--latex"])
        .output()
        .unwrap();
    assert!(output.status.success());
    String::from_utf8(output.stdout).unwrap()
}
