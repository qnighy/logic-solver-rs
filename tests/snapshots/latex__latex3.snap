---
source: tests/latex.rs
expression: "run(\"A → B → A\")"

---
\documentclass{article}
\usepackage{bussproofs}
\usepackage{colortbl}
\usepackage{varwidth}
\usepackage{pifont}
\usepackage{xcolor}
\usepackage{listings}

\newcommand{\cmark}{\textcolor{blue}{\ding{52}}}%
\newcommand{\xmark}{\textcolor{red}{\ding{56}}}%

\lstset{%
  basicstyle={\small},%
  identifierstyle={\small},%
  commentstyle={\small\itshape},%
  keywordstyle={\small\bfseries},%
  ndkeywordstyle={\small},%
  stringstyle={\small\ttfamily},
  frame={tb},
  breaklines=true,
  columns=[l]{fullflexible},%
  numbers=left,%
  xrightmargin=0em,%
  xleftmargin=3em,%
  numberstyle={\scriptsize},%
  stepnumber=1,
  numbersep=1em,%
  lineskip=-0.5ex%
}

\title{Logic solver result}

\begin{document}

\maketitle


\begin{tabular}{|c|}
  \hline
  \cellcolor[gray]{0.8}
  \begin{minipage}{10cm}
    \centering
    Result
  \end{minipage}
  \\\hline

  {
    \huge


    \cmark
    Provable

  }
  \\\hline
\end{tabular}

\section*{Input}

$A \to B \to A$




\section*{Proof}

\begin{prooftree}
\AxiomC{$[A]_{H1}$}
\RightLabel{\scriptsize${\to}_I (H2)$}
\UnaryInfC{$B \to A$}
\RightLabel{\scriptsize${\to}_I (H1)$}
\UnaryInfC{$A \to B \to A$}

\end{prooftree}




\end{document}
